src/MultiDBConfigDialog.cpp

```diff
@@ -267,56 +282,81 @@ namespace xrock_gui_model
 
     void MultiDBConfigDialog::onFinishBtnClicked()
     {
-        QString main_server_type = cbMainServerType->currentText();
-        QString main_server_path = tfMainServerPath->text();
-        QString main_server_graph = tfMainServerGraph->text();
-
-        if (main_server_path.isEmpty())
-        {
-            QMessageBox::warning(this, "Warning", "Main server path is empty!", QMessageBox::Ok);
-            return;
-        }
-        if (main_server_graph.isEmpty())
-        {
-            QMessageBox::warning(this, "Warning", "Main server graph is empty!", QMessageBox::Ok);
-            return;
-        }
         if (backends.empty())
         {
             QMessageBox::warning(this, "Warning", "Import servers are empty!", QMessageBox::Ok);
             return;
         }
 
+        QString mainServerName = cbMainServer->currentText();
+        const BackendItem &mainServer = *std::find_if(backends.begin(), backends.end(), [&](const BackendItem &b)
+                                                      { return b.name == mainServerName; });
         ConfigMap config;
-        config["main_server"]["type"] = main_server_type.toStdString();
-        config["main_server"]["graph"] = main_server_graph.toStdString();
+        config["main_server"]["type"] = mainServer.type.toStdString();
+        config["main_server"]["graph"] = mainServer.graph.toStdString();
+        config["main_server"]["name"] = mainServer.name.toStdString();
         std::string urlOrPath = config["main_server"]["type"] == "Client" ? "url" : "path";
-        config["main_server"][urlOrPath] = main_server_path.toStdString();
+        config["main_server"][urlOrPath] = mainServer.urlOrPath.toStdString();
 
-        for (const BackendItem &w : backends)
+        for (const BackendItem &b : backends)
         {
+            // skip main server (already added to config above)
+            if (b == mainServer)
+                continue;
             ConfigMap backend;
-            backend["name"] = w.name.toStdString();
-            backend["type"] = w.type.toStdString();
+            backend["name"] = b.name.toStdString();
+            backend["type"] = b.type.toStdString();
             if (backend["type"].getString() == "Client")
-                backend["url"] = w.urlOrPath.toStdString();
+                backend["url"] = b.urlOrPath.toStdString();
             else
-                backend["path"] = w.urlOrPath.toStdString();
-            backend["graph"] = w.graph.toStdString();
+                backend["path"] = b.urlOrPath.toStdString();
+            backend["graph"] = b.graph.toStdString();
             config["import_servers"].push_back(std::move(backend));
         }
-
+ 
+        config["also_lookup_main_server"] = cbLookupInMainDatabase->isChecked();
```

Comment:
This flag is not available in XDBI

Reply:
yes. as we agreed to not to have this flag , I will remove it.